### PR TITLE
Remove unused functions and clean up servlet integration test base

### PR DESCRIPTION
**Pull request by codefromthecrypt**
## What does this PR do?

This removes unused functions and clean up servlet integration test base I noticed while attempting to run WildFlyIT. Notably, when image download or container start failed, our stop function would NPE over it. The other changes are more taking care of drift or resource management.

I'm not sure what category below as it isn't a CHANGELOG worthy bug as it is test infra 🤷 

## Checklist

- [ ] This is an enhancement of existing features, or a new feature in existing plugins
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)
  - [ ] I have added tests that prove my fix is effective or that my feature works
  - [ ] Added an API method or config option? Document in which version this will be introduced
  - [ ] I have made corresponding changes to the documentation
- [ ] This is a bugfix
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)
  - [ ] I have added tests that would fail without this fix
- [ ] This is a new plugin
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)
  - [ ] My code follows the [style guidelines of this project](https://github.com/elastic/apm-agent-java/blob/main/CONTRIBUTING.md#java-language-formatting-guidelines)
  - [ ] I have made corresponding changes to the documentation
  - [ ] I have added tests that prove my fix is effective or that my feature works
  - [ ] New and existing [**unit** tests](https://github.com/elastic/apm-agent-java/blob/main/CONTRIBUTING.md#testing) pass locally with my changes
  - [ ] I have updated [supported-technologies.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/docs/supported-technologies.asciidoc)
  - [ ] Added an API method or config option? Document in which version this will be introduced
  - [ ] Added an instrumentation plugin? Describe how you made sure that old, non-supported versions are not instrumented by accident.
- [ ] This is something else
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)

### CODE REVIEW

File: integration-tests/application-server-integration-tests/src/test/java/co/elastic/apm/servlet/AbstractServletContainerIntegrationTest.java [hunk 1]
```diff
@@ -176,23 +182,10 @@ protected AbstractServletContainerIntegrationTest(AgentTestContainer.AppServer c
         container.start();
 
         if (runtimeAttachSupported()) {
-            if (AGENT_VERSION_TO_DOWNLOAD_FROM_MAVEN != null) {
```
Comment (codefromthecrypt) [inline integration-tests/application-server-integration-tests/src/test/java/co/elastic/apm/servlet/AbstractServletContainerIntegrationTest.java:179]: unnecessary condition

File: integration-tests/application-server-integration-tests/src/test/java/co/elastic/apm/servlet/AbstractServletContainerIntegrationTest.java [hunk 2]
```diff
@@ -176,23 +182,10 @@ protected AbstractServletContainerIntegrationTest(AgentTestContainer.AppServer c
...
-                container.startCliRuntimeAttach(AGENT_VERSION_TO_DOWNLOAD_FROM_MAVEN);
-            } else {
-                container.startCliRuntimeAttach(AGENT_VERSION_TO_DOWNLOAD_FROM_MAVEN);
-            }
+            container.startCliRuntimeAttach(AGENT_VERSION_TO_DOWNLOAD_FROM_MAVEN);
         }
     }
 
-    private static OkHttpClient setupHttpClient(boolean isDebug) {
```
Comment (codefromthecrypt) [inline integration-tests/application-server-integration-tests/src/test/java/co/elastic/apm/servlet/AbstractServletContainerIntegrationTest.java:187]: no longer used

File: integration-tests/application-server-integration-tests/src/test/java/co/elastic/apm/servlet/AbstractServletContainerIntegrationTest.java [hunk 3]
```diff
@@ -216,9 +209,13 @@ protected boolean runtimeAttachSupported() {
 
     @After
     public final void stopServer() {
-        container.getDockerClient()
-            .stopContainerCmd(container.getContainerId())
```
Comment (codefromthecrypt) [inline integration-tests/application-server-integration-tests/src/test/java/co/elastic/apm/servlet/AbstractServletContainerIntegrationTest.java:220]: this NPE'd on me

File: integration-tests/application-server-integration-tests/src/test/java/co/elastic/apm/servlet/AbstractServletContainerIntegrationTest.java [hunk 4]
```diff
@@ -271,29 +268,31 @@ public JsonNode assertTransactionReported(String pathToTest, int expectedRespons
...
 
-    public String executeAndValidateRequest(String pathToTest, String expectedContent, Integer expectedResponseCode,
+    public void executeAndValidateRequest(String pathToTest, String expectedContent, Integer expectedResponseCode,
                                             Map<String, String> headersMap) throws IOException, InterruptedException {
-        Response response = executeRequest(pathToTest, headersMap);
-        if (expectedResponseCode != null) {
-            assertThat(response.code())
-                .withFailMessage(response.toString() + getServerLogs())
-                .isEqualTo(expectedResponseCode);
+        final ResponseBody responseBody;
```
Comment (codefromthecrypt) [inline integration-tests/application-server-integration-tests/src/test/java/co/elastic/apm/servlet/AbstractServletContainerIntegrationTest.java:273]: unclosed resource, and also the result is no longer read

File: integration-tests/application-server-integration-tests/src/test/java/co/elastic/apm/servlet/AbstractServletContainerIntegrationTest.java [hunk 5]
```diff
@@ -271,29 +268,31 @@ public JsonNode assertTransactionReported(String pathToTest, int expectedRespons
...
                 .contains(expectedContent);
         }
-        return responseString;
     }
 
     private void executeStatusRequestAndCheckIgnored(String statusEndpoint) throws IOException {
         Map<String, String> headers = Collections.emptyMap();
-        Response response = executeRequest(statusEndpoint, headers);
-        assertThat(response.code()).isEqualTo(200);
+        try (Response response = executeRequest(statusEndpoint, headers)) {
```
Comment (codefromthecrypt) [inline integration-tests/application-server-integration-tests/src/test/java/co/elastic/apm/servlet/AbstractServletContainerIntegrationTest.java:293]: unclosed resource

File: integration-tests/application-server-integration-tests/src/test/java/co/elastic/apm/servlet/AbstractServletContainerIntegrationTest.java [hunk 6]
```diff
@@ -308,33 +307,6 @@ private void executeStatusRequestAndCheckIgnored(String statusEndpoint) throws I
 
     }
 
-    public String executeAndValidatePostRequest(String pathToTest, RequestBody postBody, String expectedContent, Integer expectedResponseCode) throws IOException, InterruptedException {
```
Comment (codefromthecrypt) [inline integration-tests/application-server-integration-tests/src/test/java/co/elastic/apm/servlet/AbstractServletContainerIntegrationTest.java:311]: no longer used

File: integration-tests/application-server-integration-tests/src/test/java/co/elastic/apm/servlet/AbstractServletContainerIntegrationTest.java [hunk 7]
```diff
@@ -529,16 +501,6 @@ private void validateMetadataEvent(JsonNode metadata) {
         assertThat(jsonContainer.get("id").textValue()).isEqualTo(container.getContainerId());
     }
 
-    private void addSpans(List<JsonNode> spans, JsonNode payload) {
```
Comment (codefromthecrypt) [inline integration-tests/application-server-integration-tests/src/test/java/co/elastic/apm/servlet/AbstractServletContainerIntegrationTest.java:532]: no longer used

File: integration-tests/application-server-integration-tests/src/test/java/co/elastic/apm/servlet/AbstractServletContainerIntegrationTest.java [hunk 8]
```diff
@@ -550,8 +512,4 @@ private void waitFor(String path) {
     public OkHttpClient getHttpClient() {
         return httpClient;
     }
-
-    public boolean isHotSpotBased() {
```
Comment (codefromthecrypt) [inline integration-tests/application-server-integration-tests/src/test/java/co/elastic/apm/servlet/AbstractServletContainerIntegrationTest.java:554]: no longer used

**Review by codefromthecrypt — COMMENTED** [review]
notes inline

**Review by jackshirazi — APPROVED** [review]

**Pull request merged**